### PR TITLE
Update generic filtering doc 

### PR DESCRIPTION
**Pull request by monicasarbu**
Update documentation for generic filtering to include details about conditions and update the examples for each Beat.

cc-ed: @dedemorton

### CODE REVIEW

File: libbeat/docs/filteringconfig.asciidoc [hunk 1]
```diff
@@ -13,55 +13,191 @@
...
+===== regexp
+
+The `regexp` condition checks the field against a regular expression. It accepts only strings.
+
+For example, the following condition checks if the process name has a certain format:
+
+[source,yaml]]
+-----
+reqexp:
+  proc.name: 58gdhsga-=kw+++
```
Comment (andrewkroh) [inline libbeat/docs/filteringconfig.asciidoc]: Just curious, what process does this regex match?

Comment (monicasarbu) [inline libbeat/docs/filteringconfig.asciidoc]: Good question. It's from our tests. I think I should replace with something more "readable" :)

**Comment by monicasarbu** [comment]
@ruflin @andrewkroh It's ready for another round of review.

File: filebeat/docs/filebeat-filtering.asciidoc [hunk 2]
```diff
@@ -29,14 +29,24 @@ See <<configuration-filebeat-options,Filebeat configuration options>> for more i
...
-with the mandatory fields.
+For example, the following filter configuration drops all the DEBUG messages.
 
-[source, yaml]
+[source,yaml]
 -----------------------------------------------------
-filters:
- - include_fields:
-     fields: ["message", "source"]
+filter:
```
Comment (ruflin) [inline filebeat/docs/filebeat-filtering.asciidoc]: This should be filters (same below)

File: libbeat/docs/filteringconfig.asciidoc [hunk 3]
```diff
@@ -13,55 +13,191 @@
...
+
+
+[source,yaml]
+------
+range:
+  gte:
+    proc.code: 300
+------
+
+NOTE: You cannot define two operators of the same time (like `gte`), but you can define two different operators under
```
Comment (ruflin) [inline libbeat/docs/filteringconfig.asciidoc]: of the same type (

File: libbeat/docs/filteringconfig.asciidoc [hunk 4]
```diff
@@ -13,55 +13,191 @@
...
+the same `range` in order to define an interval.
```
Comment (ruflin) [inline libbeat/docs/filteringconfig.asciidoc]: Not 100% sure I get this note?

File: packetbeat/docs/packetbeat-filtering.asciidoc [hunk 5]
```diff
@@ -41,4 +41,25 @@ The filtered event would look something like this:
...
 -----------------------------------------------------
 
-See <<configuration-filter,filter configuration options>> for more information.
+If you would like to drop all the successful transactions, you can use the following configuration:
+
+[source,yaml]
+------------
+filters:
+  - drop_event:
+     equals:
```
Comment (ruflin) [inline packetbeat/docs/packetbeat-filtering.asciidoc]: Indentation looks off here (also below)

**Comment by ruflin** [comment]
LGTM. Some more minor comments.

Jenkins failing is related to a flaky filebeat test which is fixed in https://github.com/elastic/beats/pull/1738

**Comment by ruflin** [comment]
LGTM. Waiting for green.

**Pull request merged**